File: app/src/main/java/com/jintao/vipmanager/database/bean/DbVipUserInfo.java
package com.jintao.vipmanager.database.bean;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Property;
import org.greenrobot.greendao.annotation.Generated;

/**
 * Author: jintao
 * CreateDate: 2023/7/14 23:08
 * Description:兑换记录，兑换时间，消费记录，消费积分，剩余积分
 */
@Entity(nameInDb = "vip_user_info")
public class DbVipUserInfo {

    @Id(autoincrement = true)
    private Long id;

    @Property(nameInDb = "user_name")
    private String userName = "";

    @Property(nameInDb = "user_sex")
    private String userSex = "";

    @Property(nameInDb = "phone_number")
    private String phoneNumber = "";

    @Property(nameInDb = "uid")
    private int uid = 0;

    //消费次数
    @Property(nameInDb = "consume_number")
    private int consumeNumber = 0;

    //最后消费
    @Property(nameInDb = "last_amount")
    private float lastAmount = 0;

    //总积分
    @Property(nameInDb = "total_integral")
    private float totalIntegral = 0;

    //可用积分
    @Property(nameInDb = "user_integral")
    private float userIntegral = 0;

    //总消费金额
    @Property(nameInDb = "total_amount")
    private float totalAmount = 0;

    //扣除消费积分后，剩余积分对应的消费金额
    @Property(nameInDb = "current_amount")
    private float currentAmount = 0;

    //最近消费时间
    @Property(nameInDb = "consume_time")
    private long consumeTime = 0;

    @Property(nameInDb = "register_time")
    private String registerTime = "";

    @Generated(hash = 1169819871)
    public DbVipUserInfo(Long id, String userName, String userSex,
            String phoneNumber, int uid, int consumeNumber, float lastAmount,
            float totalIntegral, float userIntegral, float totalAmount,
            float currentAmount, long consumeTime, String registerTime) {
        this.id = id;
        this.userName = userName;
        this.userSex = userSex;
        this.phoneNumber = phoneNumber;
        this.uid = uid;
        this.consumeNumber = consumeNumber;
        this.lastAmount = lastAmount;
        this.totalIntegral = totalIntegral;
        this.userIntegral = userIntegral;
        this.totalAmount = totalAmount;
        this.currentAmount = currentAmount;
        this.consumeTime = consumeTime;
        this.registerTime = registerTime;
    }

    @Generated(hash = 1394531373)
    public DbVipUserInfo() {
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUserName() {
        return this.userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getPhoneNumber() {
        return this.phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public float getUserIntegral() {
        return this.userIntegral;
    }

    public void setUserIntegral(float userIntegral) {
        this.userIntegral = userIntegral;
    }

    public int getUid() {
        return this.uid;
    }

    public void setUid(int uid) {
        this.uid = uid;
    }

    public String getUserSex() {
        return userSex;
    }

    public void setUserSex(String userSex) {
        this.userSex = userSex;
    }

    public int getConsumeNumber() {
        return consumeNumber;
    }

    public void setConsumeNumber(int consumeNumber) {
        this.consumeNumber = consumeNumber;
    }

    public float getLastAmount() {
        return lastAmount;
    }

    public void setLastAmount(float lastAmount) {
        this.lastAmount = lastAmount;
    }

    public float getTotalIntegral() {
        return totalIntegral;
    }

    public void setTotalIntegral(float totalIntegral) {
        this.totalIntegral = totalIntegral;
    }

    public float getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(float totalAmount) {
        this.totalAmount = totalAmount;
    }

    public float getCurrentAmount() {
        return currentAmount;
    }

    public void setCurrentAmount(float currentAmount) {
        this.currentAmount = currentAmount;
    }

    public long getConsumeTime() {
        return consumeTime;
    }

    public void setConsumeTime(long consumeTime) {
        this.consumeTime = consumeTime;
    }

    public String getRegisterTime() {
        return registerTime;
    }

    public void setRegisterTime(String registerTime) {
        this.registerTime = registerTime;
    }
}

File: app/src/main/java/com/jintao/vipmanager/database/bean/DbVipUserInfo.kt
package com.jintao.vipmanager.database.bean

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Author: jintao
 * CreateDate: 2023/7/14 23:08
 * Description:兑换记录，兑换时间，消费记录，消费积分，剩余积分
 */
@Entity(tableName = "vip_user_info")
class DbVipUserInfo {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id")
    private var id:Long? = 0L

    @ColumnInfo(name = "user_name")
    private var userName:String? = null

    @ColumnInfo(name = "user_sex")
    private var userSex:String? = null

    @ColumnInfo(name = "phone_number")
    private var phoneNumber:String? = null

    @ColumnInfo(name = "uid")
    private var uid:Int = 0

    //消费次数
    @ColumnInfo(name = "consume_number")
    private var consumeNumber:Int = 0

    //兑换次数
    @ColumnInfo(name = "exchange_number")
    private var exchangeNumber:Float = 0f

    //总积分
    @ColumnInfo(name = "total_integral")
    private var totalIntegral:Float = 0f

    //可用积分
    @ColumnInfo(name = "user_integral")
    private var userIntegral:Float = 0f

    //总消费金额
    @ColumnInfo(name = "total_amount")
    private var totalAmount:Float = 0f

    //扣除消费积分后，剩余积分对应的消费金额
    @ColumnInfo(name = "current_amount")
    private var currentAmount:Float = 0f

    //最近消费时间
    @ColumnInfo(name = "consume_time")
    private var consumeTime:Long = 0L

    @ColumnInfo(name = "register_time")
    private var registerTime:String? = null

    fun getId():Long {
        return this.id!!
    }

    fun setId(id:Long) {
        this.id = id
    }

    fun getUserName():String {
        if(this.userName==null) {
            return ""
        }else {
            return this.userName!!
        }
    }

    fun setUserName(userName:String) {
        this.userName = userName
    }

    fun getPhoneNumber():String {
        if(this.phoneNumber==null) {
            return ""
        }else {
            return this.phoneNumber!!
        }
    }

    fun setPhoneNumber(phoneNumber:String) {
        this.phoneNumber = phoneNumber
    }

    fun getUserIntegral():Float {
        return this.userIntegral
    }

    fun setUserIntegral(userIntegral:Float) {
        this.userIntegral = userIntegral
    }

    fun getUid():Int {
        return this.uid
    }

    fun setUid(uid:Int) {
        this.uid = uid
    }

    fun getUserSex():String {
        if(this.userSex==null) {
            return ""
        }else {
            return this.userSex!!
        }
    }

    fun setUserSex(userSex:String) {
        this.userSex = userSex
    }

    fun getConsumeNumber():Int {
        return consumeNumber
    }

    fun setConsumeNumber(consumeNumber:Int) {
        this.consumeNumber = consumeNumber
    }

    fun getExchangeNumber():Float {
        return exchangeNumber
    }

    fun setExchangeNumber(exchangeNumber:Float) {
        this.exchangeNumber = exchangeNumber
    }

    fun getTotalIntegral():Float {
        return totalIntegral
    }

    fun setTotalIntegral(totalIntegral:Float) {
        this.totalIntegral = totalIntegral
    }

    fun getTotalAmount():Float {
        return totalAmount
    }

    fun setTotalAmount(totalAmount:Float) {
        this.totalAmount = totalAmount
    }

    fun getCurrentAmount():Float {
        return currentAmount
    }

    fun setCurrentAmount(currentAmount:Float) {
        this.currentAmount = currentAmount
    }

    fun getConsumeTime():Long {
        return consumeTime
    }

    fun setConsumeTime(consumeTime:Long) {
        this.consumeTime = consumeTime
    }

    fun getRegisterTime():String {
        if(this.registerTime==null) {
            return ""
        }else {
            return this.registerTime!!
        }
    }

    fun setRegisterTime(registerTime:String) {
        this.registerTime = registerTime
    }
}